WooCommerce/WooCommerceTests/Model/MarkOrderAsReadUseCaseTests.swift

```diff
@@ -0,0 +1,189 @@
+import XCTest
+import WooFoundation
+@testable import Yosemite
+@testable import Networking
+@testable import Storage
+
+final class MarkOrderAsReadUseCaseTests: XCTestCase {
+    /// Mock Dispatcher!
+    ///
+    private var dispatcher: Dispatcher!
+
+    /// Mock Network: Allows us to inject predefined responses!
+    ///
+    private var network: MockNetwork!
+
+    /// Mock Storage: InMemory
+    ///
+    private var storageManager: MockStorageManager!
+
+    /// Mock stores manager
+    ///
+    private var storesManager: MockStoresManager!
+
+    /// Convenience Property: Returns the StorageType associated with the main thread.
+    ///
+    private var viewStorage: StorageType {
+        return storageManager.viewStorage
+    }
+
+    /// Sample Notes
+    ///
+    private lazy var sampleNotes: [Yosemite.Note] = {
+        return try! mapNotes(from: "notifications-load-all")
+    }()
+
+    private func sampleNote(read: Bool) -> Yosemite.Note? {
+        for note in sampleNotes {
+            if note.read == read, note.meta.identifier(forKey: .order) != nil {
+                return note
+            }
+        }
+        return nil
+    }
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockStorageManager()
+        storesManager = MockStoresManager(sessionManager: .makeForTesting())
+        network = MockNetwork()
+
+        // Need to nuke this in-between tests otherwise some will randomly fail
+        NotificationStore.resetSharedDerivedStorage()
```

Comment:
I copy/pasted it from other tests but now by looking at it more it looks like it can also be done in the tearDown, maybe even better would be to do it in both, to make sure to have clean storage before and after tests